### PR TITLE
freeze admin bounty overview section

### PR DESCRIPTION
**Pull request by Ekep-Obasi**
## Describe your changes

Freeze bounty overview on scroll using IntersectionObserver

- Intersection Observer: Used to trigger callback when scrolling past reference component.
- Custom Hook: useInViewPort encapsulates Intersection Observer logic for reusability.

Reason for Intersection Observer:

- CSS display: sticky; not applicable due to parent component's overflow property.

### Preview 

https://www.loom.com/share/bf450b3c9b6d4a778a1f2c32184b57bd?sid=7662d832-11e2-40a4-b5dc-15bf069aceec

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have tested on Chrome and Firefox
- [x] I have provided a screenshot or recording of changes in my PR if there were updates to the frontend

### CODE REVIEW

**Comment by ecurrencyhodler** [comment]
Hey @Ekep-Obasi it looks like two of your tests are failing. Can you take another look? Let us know if you need some help.

**Comment by Ekep-Obasi** [comment]
Yes @ecurrencyhodler 
 I need some help with this

**Comment by ecurrencyhodler** [comment]
@kevkevinpal @elraphty can you give some insight here?

**Comment by elraphty** [comment]
> @kevkevinpal @elraphty can you give some insight here?

@Ekep-Obasi your prettier is also failing to run ```yarn run prettier``` to fix, the issue is you added some code that increased the lines of code we have, but the tests were not increased and there was a threshold set. The error ```Jest: "./src/hooks/" coverage threshold for lines (37.03%) not met: 35.1%```, add some test to meet the coverage threshold.

**Comment by Ekep-Obasi** [comment]
@elraphty, Do you mean that I should write tests for the component I just created?

And for the prettier error in `./src/hooks/useInViewport.ts` I've ran `yarn run prettier` and even ran `npx prettier --config .prettierrc.json --check ./src/hooks/useInViewport.ts` to check. But the prettier workflow still seems to be failing 

Capture

**Comment by kevkevinpal** [comment]
> @elraphty, Do you mean that I should write tests for the component I just created?
> 
> And for the prettier error in `./src/hooks/useInViewport.ts` I've ran `yarn run prettier` and even ran `npx prettier --config .prettierrc.json --check ./src/hooks/useInViewport.ts` to check. But the prettier workflow still seems to be failing
> 
> Capture

I would create a test that covers `useInViewPort` I wouldn't worry about the prettier check as much as the jest test check

**Comment by Ekep-Obasi** [comment]
> > @elraphty, Do you mean that I should write tests for the component I just created?
> > And for the prettier error in `./src/hooks/useInViewport.ts` I've ran `yarn run prettier` and even ran `npx prettier --config .prettierrc.json --check ./src/hooks/useInViewport.ts` to check. But the prettier workflow still seems to be failing
> > Capture
> 
> I would create a test that covers `useInViewPort` I wouldn't worry about the prettier check as much as the jest test check

@kevkevinpal Thanks for the help!

**Comment by Ekep-Obasi** [comment]
@kevkevinpal @elraphty 
After adding tests for the hook and running `yarn run test`  locally all tests pass

Capture

But the jest workflow is failing with this error 

`Jest worker encountered 4 child process exceptions, exceeding retry limit`

File: frontend/app/src/pages/superadmin/header/HeaderStyles.tsx [hunk 1]
```diff
@@ -19,6 +19,11 @@ export const AlternateWrapper = styled.div`
   justify-content: space-between;
   align-items: center;
   flex-shrink: 0;
+  position: fixed;
+  top: 62px;
+  left: 0;
+  width: 100%;
+  z-index: 9999999;
```
Comment (kevkevinpal) [inline frontend/app/src/pages/superadmin/header/HeaderStyles.tsx:26]: this `z-index` doesnt feel like best practice is there a style to keep it always ontop? if you can look into alternatives to doing this that would be good

Comment (kevkevinpal) [reply frontend/app/src/pages/superadmin/header/HeaderStyles.tsx:26]: I'm just going to make this another issue so we can merge this PR for now

**Comment by ecurrencyhodler** [comment]
Tested on staging and it looks good! Paid.

**Comment by ecurrencyhodler** [comment]
Increased it to 300k for you.

**Comment by Ekep-Obasi** [comment]
@ecurrencyhodler thank you 🙏

**Pull request merged**